gridcapa-swe-commons/src/main/java/com/farao_community/farao/gridcapa_swe_commons/configuration/ProcessConfiguration.java

```diff
@@ -6,34 +6,44 @@
  */
 package com.farao_community.farao.gridcapa_swe_commons.configuration;
 
-import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.context.annotation.Configuration;
 
+import java.util.Map;
+
 /**
  * @author Marc Schwitzguébel {@literal <marc.schwitzguebel at rte-france.com>}
  */
 @Configuration
+@ConfigurationProperties(prefix = "swe-commons")
 public class ProcessConfiguration {
 
-    @Value("${swe-commons.zone-id}")
     private String zoneId;
-
-    @Value("${swe-commons.shift-max-iteration-number}")
     private Integer shiftMaxIterationNumber;
-
-    @Value("${swe-commons.modeling-authority-set}")
-    private String modelingAuthoritySet;
+    private Map<String, String> modelingAuthoritySet;
 
     public String getZoneId() {
         return zoneId;
     }
 
+    public void setZoneId(String zoneId) {
```

Comment:
It looks strange to me to add setters in a configuration class, which is supposed to reflect a configuration stored in a properties file and should not change during runtime.
Furthermore, you apparently don't use the setters you added (at least not explicitly). I advise you to remove them.

Reply:
Spring needs those setters to initialize the values

Reply:
I guess it comes from the fact that you replaced every single `@Value` with a global `@ConfigurationProperties`. Maybe a constructor would work in your case? And it would be better than setters, as it avoids runtime changes in the configuration objects.